sklearn/utils/estimator_checks.py

```diff
@@ -1495,8 +1495,38 @@ def check_classifiers_train(name, classifier_orig, readonly_memmap=False):
                 assert_array_equal(np.argsort(y_log_prob), np.argsort(y_prob))
 
 
+def check_outlier_corruption(estimator, num_outliers, expected_outliers, X):
```

Comment:
Depending on the data set this check can fail for `LocalOutlierFactor` because this estimator has no `decision_function`. The reason that it is not failing right now is because in this case (given data set, `LocalOutlierFactor` with default parameters) we have `num_outliers == expected_outliers`. I think we should only run this check for estimators that have a `decision_function`. The only estimator that does not have a `decision_function` is `LocalOutlierFactor` and we can have a specific check in `test_lof.py`.